packages/react/src/components/Workbook.tsx

```diff
@@ -0,0 +1,288 @@
+import { FlatfileClient } from '@flatfile/api'
+import type { Flatfile } from '@flatfile/api'
+import {
+  DefaultSubmitSettings,
+  ISpace,
+  JobHandler,
+  SheetHandler,
+  SimpleOnboarding,
+} from '@flatfile/embedded-utils'
+import { FlatfileEvent } from '@flatfile/listener'
+import { FlatfileRecord, recordHook } from '@flatfile/plugin-record-hook'
+import React, { JSX, useState, useContext, useEffect } from 'react'
+import { useListener } from '../hooks'
+import ConfirmModal from './ConfirmCloseModal'
+import FlatfileContext from './FlatfileContext'
+import { getContainerStyles, getIframeStyles } from './embeddedStyles'
+import SheetConfig from '@flatfile/blueprint'
+
+type IFrameTypes = Partial<
+  Pick<
+    ISpace,
+    | 'iframeStyles'
+    | 'mountElement'
+    | 'exitText'
+    | 'exitTitle'
+    | 'exitPrimaryButtonText'
+    | 'exitSecondaryButtonText'
+    | 'displayAsModal'
+    | 'closeSpace'
+  >
+>
+
+const NewSpaceContents = (
+  props: Partial<IFrameTypes> & {
+    handleCloseInstance: () => void
+  }
+): JSX.Element => {
+  const [showExitWarnModal, setShowExitWarnModal] = useState(false)
+
+  const { open, sessionSpace } = useContext(FlatfileContext)
```

Comment:
Since we are creating a net-new workbooks component here, do you think it is worth considering we incorporate the `preload` option into it? I think it would make sense for us to simply have a `preload` prop that will either load the iframe, or defer depending on the setting, rather than continuing with the idea of having 2 different components entirely